deepspeed/runtime/engine.py

```diff
@@ -2487,10 +2494,11 @@ def sparse_allreduce(self, sparse, dp_group):
 
         if self.postscale_gradients():
             if self.gradient_average:
-                values.mul_(self.gradient_predivide_factor() /
-                            (dist.get_world_size(group=dp_group) / float(self.sequence_parallel_size)))
+                if dp_world_size is None:
+                    dp_world_size = dist.get_world_size(group=dp_group)
```

Comment:
This needs to be moved to before if self.postscale_gradienst():

Reply:
fixed